doc/ipmc/IPMC_Data_Plane_HLD.md

```diff
@@ -0,0 +1,472 @@
+| IPMC            | IP Multicast                                       |
+| Mroute          | Multicast Route                                    |
+| Nhg             | NextHop Group                                      |
+
+### Overview
+
+IP multicast is a network communication technique that allows a single sender to send packets to multiple destinations without having to send packets separately for each destination. This method greatly saves bandwidth resources and improves the efficiency of the network. For different Multicast group members, Multicast service models can be divided into ASM(Any-Source multicast) and SSM(Source-Specific multicast) service models. To ensure efficient transmission of multicast data, IP multicast uses the RPF (Reverse Path Forwarding) mechanism.
+
+Routers dynamically establish forwarding tables through protocols (such as IGMP, PIM, MSDP, etc.) and maintain a multicast forwarding table, recording the membership of the multicast group and the corresponding outbound interface.
+
+Multicast packets target a specific multicast address (224.0.0.0/4 for ipv4 and FF00::/8 for ipv6).
+
+### Requirements
+
+#### Functional Requirements
+
+The following requirements are addressed by the design presented in this document:
+
+1. Support IP multicast routing: ASM or SSM
+2. Support IPv4 and IPv6 multicast routes
+3. Support IP multicast function for routing interfaces, SVI interfaces, L3 subinterfaces, and L3 Portchannel interfaces 
+4. Support non-default VRF for IP multicast routes
+5. Support warm reboot for IP multicast routes
+
+#### Configuration and Management Requirements
+
+This document proposes that configuration and management should meet the following:
+
+1. Enable IP multicast on interfaces
+
+   > The default interface is not enabled to forward multicast routes
+
+### Architecture Design
+
+This document is designed to integrate the IP multicast routing function into the SWSS Container, so that related components can support IP multicast routing installing and warm reboot functions, and finally support the IP multicast routing function at the chip level.
+
+In SONiC, unicast routes and ECMP routes are sent via the fpm component to the APPL DB for further processing on the data plane. However, for IP multicast routing, the FRR Container has not passed the IP multicast routing data to the fpm component.
+
+The frr-pimd/pim6d daemon process is introduced in the FRR Container, in order to learn IP multicast routes under the PIM protocol and install IP multicast routes to the kernel.
+
+For the implementation of the data plane, firstly, the fpm component does not support IP multicast routing right now; secondly, in order to support more IP multicast protocols as much as possible;  thirdly, the implementation and support of FRR Container are beyond the design scope of this document, so the design will use Linux kernel as multicast route source.

```

Comment:
> Similar to fpmsyncd, for IPMC can we consider using ipmcfpmsyncd instead of mroutesyncd ? This channel can also be used to get multicast enabled interface updates from kernel along with mroute entries.

The decision between using fpmsyncd or the kernel will be made after further discussion with the Routing WG. We will update the HLD accordingly.